iogt/patch.py

```diff
@@ -164,24 +163,21 @@ def store_to_db(self, pofile, locale, store_translations=False):
         else:
             locale_dir_name = get_locale_parent_dirname(pofile)
 
-        t = TranslationEntry.objects.filter(original=m.msgid, language=language, domain=domain).first()
-        if t:
-            if t.translation:
-                translations_to_keep.append(m.msgid)
-                continue
+        if tdict.get((m.msgid, language, domain)):
+            translations_to_keep.append(m.msgid)
```

Comment:
There is a case that goes like this 

1. po file is loaded that has an empty translation for a string and a DB record is created
2. po file has been updated with the translation
3. po file is loaded again and it should update the existing DB entry (only if the translation is still empty in DB)

e.g., 
locale/ar/LC_MESSAGES/django.po
msgid "Remember me"
msgstr ""

when po file is loaded DB record is created 

locale/ar/LC_MESSAGES/django.po
msgid "Remember me"
msgstr "تذكرنى"

load po file again, the same entry created in the previous step should be updated.

This refactoring is creating a new entry. I hope this makes sense but do let me know if you have any questions.

Reply:
If a new entry is created, will the old (empty) entry be removed? If so, the result would be the same as updating. It would be better not to update because that can't be done in bulk.

Maybe I should write tests to make sure we're getting the result we intend.

Reply:
Both removing old (empty) entries and creating new entries or updating old entries would work. BTW Django does provide bulk update functionality see [here](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#bulk-update). Yes, tests would be great.